File: Array & String/Roman to Integer/RomanToInteger.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;
class Solution {
 public:
  int romanToInt(string s) {
    unordered_map<char, int> mp;
    mp['I'] = 1;
    mp['V'] = 5;
    mp['X'] = 10;
    mp['L'] = 50;
    mp['C'] = 100;
    mp['D'] = 500;
    mp['M'] = 1000;
    int result = 0;
    int n = s.size();
    for (int i = 0; i < n; i++) {
      if (i < n - 1 && mp[s[i]] < mp[s[i + 1]]) {
        result += mp[s[i + 1]] - mp[s[i]];
        i++;
      } else {
        result += mp[s[i]];
      }
    }
    return result;
  }
};

int main() {
  int tc;
  cin >> tc;
  cin.ignore();
  Solution obj;
  while (tc-- > 0) {
    string s;
    getline(cin, s);
    cout << obj.romanToInt(s) << endl;
  }
  return 0;
}

File: Array & String/Roman to Integer/RomanToInteger.java
import java.util.HashMap;
import java.util.Scanner;

class Solution {
  public int romanToInt(String s) {
    HashMap<Character, Integer> mp = new HashMap<>();
    mp.put('I', 1);
    mp.put('V', 5);
    mp.put('X', 10);
    mp.put('L', 50);
    mp.put('C', 100);
    mp.put('D', 500);
    mp.put('M', 1000);
    int result = 0;
    int n = s.length();
    for (int i = 0; i < n; i++) {
      if (i < n - 1 && mp.get(s.charAt(i)) < mp.get(s.charAt(i + 1))) {
        result += mp.get(s.charAt(i + 1)) - mp.get(s.charAt(i));
        i++;
      } else {
        result += mp.get(s.charAt(i));
      }
    }
    return result;
  }
}

public class RomanToInteger {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    Solution obj = new Solution();
    int tc = sc.nextInt();
    sc.nextLine();
    while (tc-- > 0) {
      String s = sc.nextLine();
      System.out.println(obj.romanToInt(s));
    }
    sc.close();
  }
}
